app/controllers/maps_controller.rb

```diff
@@ -0,0 +1,33 @@
+class MapsController < ApplicationController
+
+  respond_to :json
+
+  def index
+  end
+
+  # def search
+  #   geometry = params['features'][0]['geometry']
+  #   @attractions = Attraction.search_within(geometry).all.to_a
+  #   render json: { attractions: @attractions }
+  # end
+
+  def search
+    geometry = params['features'][0]['geometry']
+    if current_user
+      interests = current_user.interests
+      @attractions = Attraction.search_within(geometry, interests)
```

Comment:
You could get this down to one line by using `try` and handling `nil` interests in the `search_within` method.
```ruby
@attractions = Attraction.search_within(geometry, current_user.try(:interests))
```